### PR TITLE
feat(Core): GUID recycler

### PR DESCRIPTION
**Pull request by BarbzYHOOL**
##### CHANGES PROPOSED:

-  Recycle GUIDs to avoid crashes

Fix provided by @Deku a few weeks ago.

###### ISSUES ADDRESSED:

Prevent crashes from guid limit overflow.

##### TESTS PERFORMED:

@Naxxramas-Server  (topaz) and @poszer tested but I didn't.

##### HOW TO TEST THE CHANGES:

> Deku:
> in creature template insert a creature with a high guid, almost at the limit of "int", and then go to a raid where the boss spawns many creatures those temporary creatures will overflow the guid limit and crash the server

The maximum value you can store in an signed integer is 2147483647 And unsigned int is 4294967295. I suppose we use unsigned int for these GUIDs but not sure, need to verify.

##### Target branch(es):

Master

### CODE REVIEW

**Comment by BarbzYHOOL** [comment]
It was tested already (when it was a gist) but maybe I made a typo.

File: src/server/game/Globals/ObjectMgr.cpp [hunk 1]
```diff
@@ -6341,6 +6351,31 @@ uint32 ObjectMgr::GenerateLowGuid(HighGuid guidhigh)
...
+            break;
+        }
+        case HIGHGUID_GAMEOBJECT:
+        {
+            ASSERT(_hiGoRecycledGuid < 0x00FFFFFE && "Gameobject recycled guid overflow!");
+            if (_hiGoRecycledGuid < _hiGoRecycledGuidMax)
+                return _hiGoRecycledGuid++;
+            break;
+        }
+        default: // No es manejado por el reciclador
```
Comment (Winfidonarleyan) [inline src/server/game/Globals/ObjectMgr.cpp]: What?

Comment (BarbzYHOOL) [reply src/server/game/Globals/ObjectMgr.cpp]: default case is not handled by the recycler, should I change the comment or just remove it ? (and thank you for noticing)

Comment (BarbzYHOOL) [reply src/server/game/Globals/ObjectMgr.cpp]: i translated it, i guess i can merge right?

Comment (BarbzYHOOL) [reply src/server/game/Globals/ObjectMgr.cpp]: @Winfidonarleyan feel free to merge it when you see that message :P

**Comment by BarbzYHOOL** [comment]
thank you @heyaapl for confirming

File: src/server/game/Globals/ObjectMgr.cpp (hunk 1 above)
Comment (Winfidonarleyan) [reply src/server/game/Globals/ObjectMgr.cpp]: Yes, need translating;D

**Comment by wowmane** [comment]
good job ! successful

**Pull request merged**